encoding/csv.ts

```diff
@@ -334,20 +334,12 @@ export async function readMatrix(
  * Parse the CSV string/buffer with the options provided.
  *
  * ColumnOptions provides the column definition
- * and the parse function for each entry of the
- * column.
  */
 export interface ColumnOptions {
   /**
    * Name of the column to be used as property
    */
   name: string;
-  /**
-   * Parse function for the column.
-   * This is executed on each entry of the header.
-   * This can be combined with the Parse function of the rows.
-   */
-  parse?: (input: string) => unknown;
```

Comment:
I may be reading too much into the comment thread in #881. To keep the logic consistent I removed the parse function from columns as well. Should this be done? 

Now, if the parse is removed from column options maybe the function signature for `columns?: string[] | ColumnOptions[]` should just be simplified to `columns?: string[]` as below will evaluate to the same.

```js
parse(data, { columns: ['a', 'b', 'c']});

parse(data, { columns: [{ name: 'a' }, { name: 'b' }, { name: 'c' } ] });
```

Reply:
I'm in favor of removing this